src/Parser/DemonHunter/Havoc/Modules/Spells/Nemesis.js

```diff
@@ -7,29 +7,66 @@ import Enemies from 'Parser/Core/Modules/Enemies';
 import SPELLS from 'common/SPELLS';
 import SpellIcon from 'common/SpellIcon';
 
-import { formatPercentage, formatDuration } from 'common/format';
+import calculateEffectiveDamage from 'Parser/Core/calculateEffectiveDamage';
+import { formatNumber, formatPercentage } from 'common/format';
 import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
 
+const NEMESIS_BUFF_IDS = [
+  SPELLS.NEMESIS_DEMON.id,
+  SPELLS.NEMESIS_HUMANOID.id,
+  SPELLS.NEMESIS_ABERRATION.id,
+  SPELLS.NEMESIS_BEAST.id,
+  SPELLS.NEMESIS_CRITTER.id,
+  SPELLS.NEMESIS_DRAGONKIN.id,
+  SPELLS.NEMESIS_ELEMENTAL.id,
+  SPELLS.NEMESIS_GIANT.id,
+  SPELLS.NEMESIS_MECHANICAL.id,
+  SPELLS.NEMESIS_UNDEAD.id,
+];
+
+const NEMESIS_DAMAGE_MODIFIER = 0.25;
+
 class Nemesis extends Analyzer {
   static dependencies = {
     combatants: Combatants,
     enemies: Enemies,
   };
 
+  bonusDmg = 0;
+
   on_initialized() {
     this.active = this.combatants.selected.hasTalent(SPELLS.NEMESIS_TALENT.id);
   }
 
-  statistic() {
-    const nemesisUptime = this.enemies.getBuffUptime(SPELLS.NEMESIS_TALENT.id);
-    const nemesisUptimePercentage = nemesisUptime / this.owner.fightDuration;
+  get hasNemesisBuff() {
+    const buffs = this.combatants.selected.activeBuffs();
+    return buffs.some(buff => NEMESIS_BUFF_IDS.includes(buff.ability.guid));
+  }
 
+  on_byPlayer_damage(event) {
+    if(event.targetIsFriendly) {
+      return;
+    }
+    const enemy = this.enemies.getEntity(event);
+    if(enemy.hasBuff(SPELLS.NEMESIS_TALENT.id) || this.hasNemesisBuff) {
+      this.bonusDmg += calculateEffectiveDamage(event, NEMESIS_DAMAGE_MODIFIER);
```

Comment:
It looks to me as though you aren't actually checking for matched creature type here... isn't that a fairly large oversight?

Reply:
![image](https://user-images.githubusercontent.com/19510201/35832696-e1e36714-0a93-11e8-8008-540daa526617.png)
I dont think you can get the enemy type from the log =/

Reply:
![image](https://user-images.githubusercontent.com/29204244/35832880-2c5cb92c-0acf-11e8-92de-317fb0855e8e.png)

Reply:
nah
![image](https://user-images.githubusercontent.com/19510201/35834161-fab8ce18-0a99-11e8-9627-b5ebf229ce4e.png)